tests/vmcheck/run-one-test.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# Wrap a single test-foo.sh by launching a VM,
+# and grabbing the journal etc. if it fails.
+
+set -euo pipefail
+
+tf=$1
+
+dn=$(cd $(dirname $0) && pwd)
+. ${commondir}/libvm.sh
+
+LOG=${LOG:-vmcheck.log}
+LOG=$(realpath $LOG)
+
+# NB: allow JOURNAL_LOG to be empty, which means we never
+# fetch it
+JOURNAL_LOG=${JOURNAL_LOG-vmcheck-journal.txt}
+if [ -n "$JOURNAL_LOG" ]; then
+  JOURNAL_LOG=$(realpath $JOURNAL_LOG)
+fi
+
+echo "Beginning vmcheck TESTS=${TESTS:-all}"
+
+set -x
+
+# Ensure we have a connection to VM, and overlay our built
+# binaries.
+vm_setup ${tf}
+${dn}/overlay.sh
+
+# Sanity check connection
+if ! vm_ssh_wait 30; then
+  echo "ERROR: A running VM is required for 'make vmcheck'."
+  exit 1
+fi
+
+echo "VM is running."
+
+# remember the csum we're currently on and tag it so that ostree doesn't wipe it
+csum_orig=$(vm_get_booted_csum)
+vm_cmd ostree rev-parse $csum_orig &> /dev/null # validate
+vm_cmd ostree refs vmcheck_orig --delete
+vm_cmd ostree refs $csum_orig --create vmcheck_orig
+
+# Santiy check we're on vmcheck
+origin=$(vm_get_booted_deployment_info origin)
+test "$origin" = "vmcheck"
+
+echo "Neutering /etc/yum.repos.d"
+# Move the current one to .bak
+vm_cmd mv /etc/yum.repos.d{,.bak}
+vm_cmd mkdir /etc/yum.repos.d
+
+if vm_cmd test -f /usr/etc/rpm-ostreed.conf; then
+    vm_cmd cp -f /usr/etc/rpm-ostreed.conf /etc
+    # Unless we're doing overrides
+    if [[ "${VMCHECK_FLAGS:-}" =~ "stage-deployments" ]]; then
+        vm_cmd 'echo "[Experimental]" >> /etc/rpm-ostreed.conf'
+        vm_cmd 'echo StageDeployments=true >> /etc/rpm-ostreed.conf'
+    fi
+fi
+
+origdir=$(pwd)
+echo -n '' > ${LOG}
+
+testdir="$(dirname $(realpath $0))"
+cd $testdir
+
+bn=$(basename ${tf})
+echo "Running $bn..."
+printf "\n==== ${bn} ====\n" >> ${LOG}
+vm_cmd logger "vmcheck: running $bn..."
+
+if ! ${dn}/${tf}; then
+    echo "FAILED: ${tf}"
+    vm_cmd 'journalctl --no-pager || true' > ${JOURNAL_LOG} || true
+    if test -z "${TEST_DEBUG:-}" &&
+       test -n "${VMCHECK_TMPD:-}" &&
```

Comment:
Is this something set by a human, or supposed to be set in the STI path? I don't see it set anywhere.